commands/init.js

```diff
@@ -174,36 +171,36 @@ module.exports = argv => {
     if (isComponents(folderName, packageName, appType)) {
       setTarget(appType, target);
       console.log(green, `Start creating project for Components type, target: ${target}`, normal);
+      templatePath = path.join(__dirname, '../template_multiplatform_components');
+      copyComponentsFiles(templatePath, `${currentDirectory}/${folderName}`);
       if (typeScript) {
-        templatePath = path.join(__dirname, '../template_components');
+        fs.unlinkSync(`${currentDirectory}/${folderName}/lumin/src/main.js`);
+        fs.unlinkSync(`${currentDirectory}/${folderName}/src/app.js`);
+        // Copy typescript template overlay over existing template files
+        templatePath = `${__dirname}/../template_overlay_typescript_components`;
         copyFiles(templatePath, `${currentDirectory}/${folderName}`);
-        console.log(green, `Project successfully created for platforms: ${target}`, normal);
-      } else {
-        templatePath = path.join(__dirname, '../template_multiplatform_components');
-        copyComponentsFiles(templatePath, `${currentDirectory}/${folderName}`);
-        copyManifest(`${currentDirectory}/${folderName}`);
-        util.renameComponentsFiles(folderName, packageName, visibleName);
-        try {
-          fs.symlinkSync(`../resources`, `${currentDirectory}/${folderName}/reactnative/resources`, 'dir');
-        } catch (error) {
-          console.log(yellow, `Couldn't create symlink for resources directory. Please do it manually if you want to use resources in your project. For more information check: https://magicscript.org/`);
-        }
-        preparePlatforms(`${currentDirectory}/${folderName}`);
-        console.log(green, `Project successfully created for platforms: ${target}`, normal);
       }
+      copyManifest(`${currentDirectory}/${folderName}`);
+      util.renameComponentsFiles(folderName, packageName, visibleName);
+      try {
+        fs.symlinkSync(`../resources`, `${currentDirectory}/${folderName}/reactnative/resources`, 'dir');
+      } catch (error) {
+        console.log(yellow, `Couldn't create symlink for resources directory. Please do it manually if you want to use resources in your project. For more information check: https://magicscript.org/`);
+      }
+      preparePlatforms(`${currentDirectory}/${folderName}`);
```

Comment:
This function name is not correct.  You are removing not selected platforms.  A better choice would be `prunePlatforms`.  Also this function should be called before you call `renameComponentsFiles `.  This means that you need to have the symlink creation to check whether we have a `reactnative` folder or not.

Reply:
@panwrona please create a new issue for this and fix it after this is landed